Sources/Testing/Running/EntryPoint.swift

```diff
@@ -315,42 +496,69 @@ private func _eventHandlerForStreamingEvents(toFileAtPath path: String) throws -
+    func isASCIINewline(_ byte: UInt8) -> Bool {
+      byte == 10 || byte == 13
+    }
 
 #if DEBUG
-      // We don't actually expect the JSON encoder to produce output containing
-      // newline characters, so in debug builds we'll log a diagnostic message.
-      if snapshotJSON.contains(where: isASCIINewline) {
-        let message = Event.ConsoleOutputRecorder.warning(
-          "JSON encoder produced one or more newline characters while encoding an event snapshot with kind '\(event.kind)'. Please file a bug report at https://github.com/apple/swift-testing/issues/new",
-          options: .for(.stderr)
-        )
+    // We don't actually expect the JSON encoder to produce output containing
+    // newline characters, so in debug builds we'll log a diagnostic message.
+    if eventAndContextJSON.contains(where: isASCIINewline) {
+      let message = Event.ConsoleOutputRecorder.warning(
+        "JSON encoder produced one or more newline characters while encoding an event snapshot. Please file a bug report at https://github.com/apple/swift-testing/issues/new",
+        options: .for(.stderr)
+      )
 #if SWT_TARGET_OS_APPLE
-        try? FileHandle.stderr.write(message)
+      try? FileHandle.stderr.write(message)
 #else
-        print(message)
+      print(message)
+#endif
+    }
 #endif
+
+    // Remove newline characters to conform to JSON lines specification.
+    var eventAndContextJSON = Array(eventAndContextJSON)
+    eventAndContextJSON.removeAll(where: isASCIINewline)
+
+    try? file.withLock {
+      try eventAndContextJSON.withUnsafeBytes { eventAndContextJSON in
+        try file.write(eventAndContextJSON)
       }
+      try file.write("\n")
+    }
+  }
+}
 #endif
 
-      // Remove newline characters to conform to JSON lines specification.
-      var snapshotJSON = Array(snapshotJSON)
-      snapshotJSON.removeAll(where: isASCIINewline)
-      if !snapshotJSON.isEmpty {
-        try? file.withLock {
-          try snapshotJSON.withUnsafeBytes { snapshotJSON in
-            try file.write(snapshotJSON)
-          }
-          try file.write("\n")
-        }
+/// Create an event handler that encodes events as JSON and forwards them to an
+/// ABI-friendly event handler.
+///
+/// - Parameters:
+///   - eventHandler: The event handler to forward events to. See
+///     ``ABIEntryPoint_v0`` for more information.
+///
+/// - Returns: An event handler.
+///
+/// The resulting event handler outputs data as JSON. For each event handled by
+/// the resulting event handler, a JSON object representing it and its
+/// associated context is created and is passed to `eventHandler`. These JSON
+/// objects are guaranteed not to contain any ASCII newline characters (`"\r"`
+/// or `"\n"`).
+///
+/// Note that `_eventHandlerForStreamingEvents(toFileAtPath:)` calls this
+/// function and performs additional postprocessing before writing JSON data.
+private func _eventHandlerForStreamingEvents(
```

Comment:
Is this function coupled to the specific ABI signature, and if so, should it be suffixed with "_v0" to match other things which are?

Reply:
This one's private, fortunately, so the name doesn't need to represent future functionality. In a hypothetical future where we introduce a "v1" variant of everything, we'd probably rename this one "_v0" at that point, yeah.

I can rename it now without harm, of course, just wasn't critical because nothing outside the file can see it. :)

Reply:
Yeah, I realized it was private, just thought it might help us understand/remember/reinforce the fact that it is currently v0-specific if it had that suffix now

Reply:
Looks like you did this, thanks 👍